### PR TITLE
Sherwani/add demo html capture entity

### PR DESCRIPTION
**Pull request by HussainKhanSherwani**
# Pull Request Template

Before submitting this PR, please make sure you have completed the following checklist. This helps ensure that our codebase remains high quality and that our features work well across all supported devices and themes.

### Code Review
- [ ] I have carefully reviewed my code and believe it is final from my side.
- [ ] I have ensured there is no hardcoded color in my changes. Colors should be defined in theme files and reused.

### Theme Compatibility
- [ ] I have tested the new feature or updates in both light and dark themes.
- [ ] I confirm that the visual aspects of the feature are compatible and appealing in both themes.

### Device Compatibility
- [ ] I have tested my changes on mobile screen sizes.
- [ ] I have tested my changes on tablet screen sizes.
- [ ] I have tested my changes on laptop screen sizes.

### Functional Testing
- [ ] I have thoroughly tested the functionality of my changes.
- [ ] I am confident that my changes do not negatively impact existing features.

### Testing Documentation
Please provide a brief description of the tests you have performed to verify your changes. Include any relevant scenarios or edge cases.

- Example test 1: Tested feature X by uploading the image
- Example test 2: Tested feature X by doing these things.

# Demo Video
Please provide a demo video of your changes. This helps reviewers understand the changes and verify the functionality.

You can record the video using tools like Loom(https://www.loom.com/), Clip by ClickUp(https://clickup.com/features/clips), Sendspark(https://www.sendspark.com/), etc.

Include the link to the video below and make sure the video covers
- [ ] Explain the task/bug you are solving
- [ ] Show the changes in the UI
- [ ] Show the changes in the codebase
- [ ] Show the changes you tested to make sure it works
- [ ] Show the changes in various screen sizes
- [ ] Show the changes in both light and dark themes

### CODE REVIEW

File: academy-ui/prisma/schema.prisma [hunk 1]
```diff
@@ -359,6 +359,14 @@ model ClickableDemos {
...
 }
 
+model ClickableDemoHtmlCaptures{
+  id              String   @id @db.VarChar(255)
+  clickableDemoId String   @map("clickable_demo_id") @db.VarChar(255)
+  fileName        String   @map("file_name") @db.VarChar(1024)
+  fileUrl         String   @map("file_url") @db.VarChar(1024)
+  imageUrl        String   @map("image_url") @db.VarChar(1024)
+  createdAt       DateTime @default(now()) @map("created_at")
+}
```
Comment (RobinNagpal) [inline academy-ui/prisma/schema.prisma:369]: use `@@map and a consistent name for the table, just like we have for other tabels`

File: academy-ui/src/graphql/clickableDemoHtmlCapture.graphql [hunk 2]
```diff
@@ -0,0 +1,14 @@
+fragment ClickableDemoHtmlCapture on ClickableDemoHtmlCapture{
```
Comment (RobinNagpal) [inline academy-ui/src/graphql/clickableDemoHtmlCapture.graphql:1]: We don't need this now, as manually add Request and Response types now. See https://github.com/RobinNagpal/dodao-ui/blob/main/docs/RequestAndResponseTypes.md

File: academy-ui/prisma/schema.prisma [hunk 3]
```diff
@@ -359,6 +359,14 @@ model ClickableDemos {
   @@map("clickable_demos")
 }
 
+model ClickableDemoHtmlCaptures{
+  id              String   @id @db.VarChar(255)
+  clickableDemoId String   @map("clickable_demo_id") @db.VarChar(255)
+  fileName        String   @map("file_name") @db.VarChar(1024)
+  fileUrl         String   @map("file_url") @db.VarChar(1024)
+  imageUrl        String   @map("image_url") @db.VarChar(1024)
```
Comment (RobinNagpal) [inline academy-ui/prisma/schema.prisma:367]: rename to file_image_url for more clarity

File: academy-ui/src/app/api/[spaceId]/actions/html-captures/route.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  const args: MutationCreateClickableDemoHtmlCaptureArgs = await req.json();
+  const { searchParams } = new URL(req.url);
+  const spaceId = searchParams.get('spaceId');
+  const apiKey = req.headers.get('X-API-KEY');
+  const spaceById = await getSpaceById(spaceId!);
+  console.log('spaceById', spaceById);
+  try {
+    if (apiKey) {
+      await validateApiKey(apiKey, spaceId!);
+    } else if (spaceById.id !== 'tidbitshub') {
```
Comment (RobinNagpal) [inline academy-ui/src/app/api/[spaceId]/actions/html-captures/route.ts:22]: This condition `(spaceById.id !== 'tidbitshub')` is not needed here. Only if else is fine. Also may be just the API key check in the post handler as this will not be called by the backend

File: academy-ui/src/schema.graphql [hunk 5]
```diff
@@ -376,7 +376,14 @@ type ClickableDemo {
   title: String!
   updatedAt: DateTimeISO!
 }
-
+type ClickableDemoHtmlCapture{
```
Comment (RobinNagpal) [inline academy-ui/src/schema.graphql:379]: Please revert all the graphql changes

File: academy-ui/src/app/api/[spaceId]/actions/html-captures/route.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        fileName: args.input.fileName,
+        fileUrl: args.input.fileUrl,
+        fileImageUrl: args.input.fileImageUrl,
+        createdAt: new Date(),
+      },
+    });
+    return NextResponse.json({ capture }, { status: 200 });
+  } catch (error) {
+    await logError((error as any)?.response?.data || 'Error in createHtmlCapture', {}, error as any, null, null);
+    throw new Error(`Failed to create ClickableDemoHtmlCapture: ${(error as Error).message}`);
```
Comment (RobinNagpal) [inline academy-ui/src/app/api/[spaceId]/actions/html-captures/route.ts:38]: We should return error here?

File: academy-ui/src/graphql/generated/generated-types.ts [hunk 7]
```diff
@@ -11923,4 +11924,4 @@ export function useUpsertTimelineMutation(baseOptions?: Apollo.MutationHookOptio
       }
 export type UpsertTimelineMutationHookResult = ReturnType<typeof useUpsertTimelineMutation>;
 export type UpsertTimelineMutationResult = Apollo.MutationResult<UpsertTimelineMutation>;
-export type UpsertTimelineMutationOptions = Apollo.BaseMutationOptions<UpsertTimelineMutation, UpsertTimelineMutationVariables>;
+export type UpsertTimelineMutationOptions = Apollo.BaseMutationOptions<UpsertTimelineMutation, UpsertTimelineMutationVariables>;
```
Comment (RobinNagpal) [inline academy-ui/src/graphql/generated/generated-types.ts:11927]: Do we need this change?
